include/pistache/async.h

```diff
@@ -188,6 +189,7 @@ namespace Async {
             std::mutex mtx;
             std::vector<std::shared_ptr<Request>> requests;
             TypeId id;
+            bool is_construct;
```

Comment:
Since we have `std::forward` and unless I didn't see something, we can certainly simply write

```cpp
bool is_construct = false;
```

Reply:
xqing2003: Do you have any thoughts on the feedback from milipili?

Can you author a unit-test the triggers the memory leak, so that I can do a before/after test nf the proposed fix?

I still need to fix the SIGINT handler so that pistache programs that call Pistache::Http::Endpoint::serve() will exit cleanly.  Without that, valgrind will report leaks.